src/features/terminal/terminalEnvVarInjector.ts

```diff
@@ -110,7 +110,6 @@ export class TerminalEnvVarInjector implements Disposable {
                 await this.injectEnvironmentVariablesForWorkspace(workspaceFolder);
             } else {
                 // No provided workspace - update all workspaces
-                this.envVarCollection.clear();
```

Comment:
I dont know about this one though, I think we should still clear if in no workspace scenario.